manager/integration/tests/test_ha.py

```diff
@@ -2253,7 +2276,87 @@ def test_reuse_failed_replica_with_scheduling_check():
     11. Verify the failed replica on node1 is reused.
     12. Verify the volume r/w still works fine.
     """
-    pass
+    long_wait = 60*60
+    replenish_wait_setting = \
+        client.by_id_setting(SETTING_REPLICA_REPLENISHMENT_WAIT_INTERVAL)
+    client.update(replenish_wait_setting, value=str(long_wait))
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    client.update(replica_node_soft_anti_affinity_setting, value="false")
+
+    nodes = client.list_node()
+    tags = ["avail"]
+    for node in nodes:
+        set_tags_for_node_and_its_disks(client, node, tags)
+
+    client.create_volume(name=volume_name, size=SIZE, numberOfReplicas=3,
+                         diskSelector=tags, nodeSelector=tags)
+    vol = common.wait_for_volume_detached(client, volume_name)
+    assert vol.diskSelector == tags
+    assert vol.nodeSelector == tags
+    vol.attach(hostId=get_self_host_id())
+    vol = common.wait_for_volume_healthy(client, volume_name)
+    data = {
+        'pos': 0,
+        'content': common.generate_random_data(16*Ki),
+    }
+    common.write_volume_data(vol, data)
+
+    nodes = client.list_node()
+    assert len(nodes) == 3
+    node_1, node_2, node_3 = nodes
+    common.set_node_scheduling(client, node_1, False)
+    common.set_node_scheduling(client, node_2, False)
+
+    vol = client.by_id_volume(volume_name)
+    replica_1, replica_2, replica_3 = None, None, None
+    for r in vol.replicas:
+        if r.hostId == node_1.name:
+            replica_1 = r
+        elif r.hostId == node_2.name:
+            replica_2 = r
+        elif r.hostId == node_3.name:
+            replica_3 = r
+    assert len({replica_1, replica_2, replica_3}) == 3
```

Comment:
If there is a `None` replica, the result is still 3. Hence it's better to verify each replica is not None one by one.

Reply:
Thank you. Fixed